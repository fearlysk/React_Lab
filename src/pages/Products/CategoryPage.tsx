import { useParams } from "react-router-dom";
import { useState, useEffect } from "react";
import IProduct from "@/interfaces/IProduct";
import ProductsCard from "./ProductsCard/ProductsCard";
import styles from "./Products.module.scss";
import fetchData from "../../api/fetchData";

function CategoryPage() {
  const { category } = useParams();
  const [products, setProducts] = useState([]);

  useEffect(() => {
    fetchData(`http://localhost:3000/products?${category}=true`).then((data) => setProducts(data));
  }, []);

  return (
    <div>
      <h2 className={styles.headline}>Products for: {category?.toUpperCase()}</h2>
      <div className={styles.productsList}>
        {products.map((product: IProduct) => (
          <ProductsCard key={product.id} {...product} />
        ))}
      </div>
    </div>
  );
}

export default CategoryPage;
