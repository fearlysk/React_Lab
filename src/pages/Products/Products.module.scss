@import "../../styles/parts/_colors.scss";

.headline {
  margin: 15px 20px 0 20px;
  padding: 0 0 10px 0;
  border-bottom: 1px solid white;
}
.wrapper {
  display: flex;
  flex-direction: column;
  width: 80%;
  margin: 10px auto;
  border: 1px solid $purple;
  border-radius: 10px;
  background-color: rgba(120, 120, 120, 0.16);
}
.productsList {
  display: flex;
  flex-direction: row;
  justify-content: center;
  flex-wrap: wrap;
  padding: 0 0 25px 0;
}
.hidden {
  display: none;
}
