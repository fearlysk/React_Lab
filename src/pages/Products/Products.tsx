import { useEffect, useState } from "react";
import { useParams } from "react-router-dom";
import classNames from "classnames";
import ProductsCard from "./ProductsCard/ProductsCard";
import Loader from "../../components/UI/Loader/Loader";
import styles from "./Products.module.scss";
import IProduct from "@/interfaces/IProduct";
import { getProducts } from "../../api/products";
import settlePromises from "../../utils/settlePromises";
import objectToGetParams from "../../utils/urls";
import sortingOrders from "../../enums/sorting";

function Products() {
  const { category } = useParams();
  const [value, setValue] = useState("");
  const [products, setProducts] = useState([]);
  const [productsTemp, setProductsTemp] = useState([]);
  const [categoryProductsTemp, setCategoryProductsTemp] = useState([]);
  const [categoryProducts, setCategoryProducts] = useState([]);
  const [isLoading, setIsLoading] = useState(false);

  const query = objectToGetParams({ category });
  const promises = [getProducts(), getProducts(query)];

  const filteredProducts = products.filter((product: IProduct) =>
    product.title.toLowerCase().includes(value.toLowerCase())
  );

  const setLoading = () => {
    setIsLoading(true);
    setTimeout(() => setIsLoading(false), 500);
  };

  const sortProducts = (option: never, order: string) => {
    setLoading();
    if (order === sortingOrders.ASC) {
      const sortedByAsc = products.sort((a: IProduct, b: IProduct) => a[option] - b[option]).slice();
      const sortedCategoryByAsc = categoryProducts.sort((a: IProduct, b: IProduct) => a[option] - b[option]).slice();

      setProducts(sortedByAsc);
      setCategoryProducts(sortedCategoryByAsc);
    } else if (order === sortingOrders.DESC) {
      const sortedByDesc = products.sort((a: IProduct, b: IProduct) => b[option] - a[option]).slice();
      const sortedCategoryByDesc = categoryProducts.sort((a: IProduct, b: IProduct) => b[option] - a[option]).slice();

      setProducts(sortedByDesc);
      setCategoryProducts(sortedCategoryByDesc);
    }
  };

  const filterByGenre = (genre?: string) => {
    setLoading();
    if (!genre) {
      setProducts(productsTemp);
      setCategoryProducts(categoryProductsTemp);
    } else {
      setProducts(productsTemp);
      setCategoryProducts(categoryProductsTemp);

      const filteredByGenre = productsTemp.filter((item: IProduct) => item.genre === genre).slice();
      const filteredCategoryByGenre = categoryProductsTemp.filter((item: IProduct) => item.genre === genre).slice();

      setProducts(filteredByGenre);
      setCategoryProducts(filteredCategoryByGenre);
    }
  };

  const filterByAge = (age: number) => {
    setLoading();
    setProducts(productsTemp);
    setCategoryProducts(categoryProductsTemp);

    const filteredByAge = productsTemp.filter((item: IProduct) => item.age <= age).slice();
    const filteredCategoryByAge = categoryProductsTemp.filter((item: IProduct) => item.age <= age).slice();

    setProducts(filteredByAge);
    setCategoryProducts(filteredCategoryByAge);
  };

  const [selectedRadioBtn, setSelectedRadioBtn] = useState(0);
  const selectRadio = (val: number): boolean => selectedRadioBtn === val;
  const handleRadioClick = (e: React.ChangeEvent<HTMLInputElement>): void => {
    setSelectedRadioBtn(parseInt(e.currentTarget.value, 10));
    filterByAge(parseInt(e.currentTarget.value, 10));
  };

  useEffect(() => {
    setLoading();
    settlePromises(promises).then((data) => {
      setProducts(data[0] as never);
      setProductsTemp(data[0] as never);
      setCategoryProducts(data[1] as never);
      setCategoryProductsTemp(data[1] as never);
    });
  }, []);

  return (
    <div>
      {isLoading && <Loader />}
      <div className={styles.searchForm}>
        <h2 className={styles.searchFormHeadline}>Search by name:</h2>
        <input
          type="text"
          placeholder="Search..."
          className={styles.searchInput}
          onChange={(event) => setValue(event.target.value)}
        />
      </div>
      <div className={styles.body}>
        <div className={styles.filtration}>
          <h1 className={styles.headline}>Filter Products</h1>
          <div className={styles.filtrationItem}>
            <h2 className={styles.headline}>Price: </h2>
            <button
              className={styles.filterOption}
              type="button"
              onClick={() => sortProducts("price" as never, sortingOrders.ASC)}
            >
              ASC
            </button>
            <button
              className={styles.filterOption}
              type="button"
              onClick={() => sortProducts("price" as never, sortingOrders.DESC)}
            >
              DESC
            </button>
          </div>
          <div className={styles.filtrationItem}>
            <h2 className={styles.headline}>Rating: </h2>
            <button
              className={styles.filterOption}
              type="button"
              onClick={() => sortProducts("rating" as never, sortingOrders.ASC)}
            >
              ASC
            </button>
            <button
              className={styles.filterOption}
              type="button"
              onClick={() => sortProducts("rating" as never, sortingOrders.DESC)}
            >
              DESC
            </button>
          </div>
          <div className={styles.filtrationItem}>
            <h2 className={styles.headline}>Genre: </h2>
            <button className={styles.filterOption} type="button" onClick={() => filterByGenre("Action")}>
              Action
            </button>
            <button className={styles.filterOption} type="button" onClick={() => filterByGenre("Shooter")}>
              Shooter
            </button>
            <button className={styles.filterOption} type="button" onClick={() => filterByGenre()}>
              All
            </button>
          </div>
          <div className={styles.filtrationItem}>
            <h2 className={styles.headline}>Age: </h2>
            <div className={styles.filterOptionRadio}>
              <label htmlFor="12">
                12+
                <input type="radio" name="12" value="12" checked={selectRadio(12)} onChange={handleRadioClick} />
              </label>
            </div>
            <div className={styles.filterOptionRadio}>
              <label htmlFor="12">
                16+
                <input type="radio" name="16" value="16" checked={selectRadio(16)} onChange={handleRadioClick} />
              </label>
            </div>
            <div className={styles.filterOptionRadio}>
              <label htmlFor="12">
                18+
                <input type="radio" name="16" value="18" checked={selectRadio(18)} onChange={handleRadioClick} />
              </label>
            </div>
          </div>
        </div>
        <div className={styles.wrapper}>
          <div className={classNames(styles.productsWrapper, { [styles.hidden]: !value })}>
            <h2 className={styles.headline}>Search: </h2>
            <div className={styles.productsList}>
              {filteredProducts.map((product: IProduct) => (
                <ProductsCard key={product.id} {...product} />
              ))}
              {!filteredProducts.length ? <h2 className={styles.headline}>Product not found</h2> : null}
            </div>
          </div>
          {category ? (
            <div>
              <h2 className={styles.headline}>Products for: {category?.toUpperCase()}</h2>
              <div className={styles.productsList}>
                {categoryProducts.map((sortedProduct: IProduct) => (
                  <ProductsCard key={sortedProduct.id} {...sortedProduct} />
                ))}
                {categoryProducts.length === 0 && <h2 className={styles.headline}>No results found</h2>}
              </div>
            </div>
          ) : (
            <>
              <h2 className={styles.headline}>All Products</h2>
              <div className={styles.productsList}>
                {products.map((product: IProduct) => (
                  <ProductsCard key={product.id} {...product} />
                ))}
              </div>
            </>
          )}
        </div>
      </div>
    </div>
  );
}

export default Products;
