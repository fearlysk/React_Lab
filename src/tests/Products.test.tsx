import { render } from "@testing-library/react";
import Products from "../pages/Products/Products";

describe("Products component tests", () => {
  it("Render products component", () => {
    render(<Products />);
  });

  it("Check if label 'Filter Products' is in the document", () => {
    const { getByText } = render(<Products />);
    expect(getByText("Filter Products")).toBeInTheDocument();
  });

  it("Check if products page matches recent snapshot", () => {
    const products = render(<Products />);
    expect(products).toMatchSnapshot();
  });
});
