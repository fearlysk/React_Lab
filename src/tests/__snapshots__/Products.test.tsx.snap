// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Products component tests Check if products page matches recent snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="ldsRoller"
        >
          <div />
          <div />
          <div />
          <div />
          <div />
          <div />
          <div />
          <div />
        </div>
        <div
          class="searchForm"
        >
          <h2
            class="searchFormHeadline"
          >
            Search by name:
          </h2>
          <input
            class="searchInput"
            placeholder="Search..."
            type="text"
          />
        </div>
        <div
          class="body"
        >
          <div
            class="filtration"
          >
            <h1
              class="headline"
            >
              Filter Products
            </h1>
            <div
              class="filtrationItem"
            >
              <h2
                class="headline"
              >
                Price: 
              </h2>
              <button
                class="filterOption"
                type="button"
              >
                ASC
              </button>
              <button
                class="filterOption"
                type="button"
              >
                DESC
              </button>
            </div>
            <div
              class="filtrationItem"
            >
              <h2
                class="headline"
              >
                Rating: 
              </h2>
              <button
                class="filterOption"
                type="button"
              >
                ASC
              </button>
              <button
                class="filterOption"
                type="button"
              >
                DESC
              </button>
            </div>
            <div
              class="filtrationItem"
            >
              <h2
                class="headline"
              >
                Genre: 
              </h2>
              <button
                class="filterOption"
                type="button"
              >
                Action
              </button>
              <button
                class="filterOption"
                type="button"
              >
                Shooter
              </button>
              <button
                class="filterOption"
                type="button"
              >
                All
              </button>
            </div>
            <div
              class="filtrationItem"
            >
              <h2
                class="headline"
              >
                Age: 
              </h2>
              <div
                class="filterOptionRadio"
              >
                <label
                  for="12"
                >
                  12+
                  <input
                    name="12"
                    type="radio"
                    value="12"
                  />
                </label>
              </div>
              <div
                class="filterOptionRadio"
              >
                <label
                  for="12"
                >
                  16+
                  <input
                    name="16"
                    type="radio"
                    value="16"
                  />
                </label>
              </div>
              <div
                class="filterOptionRadio"
              >
                <label
                  for="12"
                >
                  18+
                  <input
                    name="16"
                    type="radio"
                    value="18"
                  />
                </label>
              </div>
            </div>
          </div>
          <div
            class="wrapper"
          >
            <div
              class="productsWrapper hidden"
            >
              <h2
                class="headline"
              >
                Search: 
              </h2>
              <div
                class="productsList"
              >
                <h2
                  class="headline"
                >
                  Product not found
                </h2>
              </div>
            </div>
            <h2
              class="headline"
            >
              All Products
            </h2>
            <div
              class="productsList"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="ldsRoller"
      >
        <div />
        <div />
        <div />
        <div />
        <div />
        <div />
        <div />
        <div />
      </div>
      <div
        class="searchForm"
      >
        <h2
          class="searchFormHeadline"
        >
          Search by name:
        </h2>
        <input
          class="searchInput"
          placeholder="Search..."
          type="text"
        />
      </div>
      <div
        class="body"
      >
        <div
          class="filtration"
        >
          <h1
            class="headline"
          >
            Filter Products
          </h1>
          <div
            class="filtrationItem"
          >
            <h2
              class="headline"
            >
              Price: 
            </h2>
            <button
              class="filterOption"
              type="button"
            >
              ASC
            </button>
            <button
              class="filterOption"
              type="button"
            >
              DESC
            </button>
          </div>
          <div
            class="filtrationItem"
          >
            <h2
              class="headline"
            >
              Rating: 
            </h2>
            <button
              class="filterOption"
              type="button"
            >
              ASC
            </button>
            <button
              class="filterOption"
              type="button"
            >
              DESC
            </button>
          </div>
          <div
            class="filtrationItem"
          >
            <h2
              class="headline"
            >
              Genre: 
            </h2>
            <button
              class="filterOption"
              type="button"
            >
              Action
            </button>
            <button
              class="filterOption"
              type="button"
            >
              Shooter
            </button>
            <button
              class="filterOption"
              type="button"
            >
              All
            </button>
          </div>
          <div
            class="filtrationItem"
          >
            <h2
              class="headline"
            >
              Age: 
            </h2>
            <div
              class="filterOptionRadio"
            >
              <label
                for="12"
              >
                12+
                <input
                  name="12"
                  type="radio"
                  value="12"
                />
              </label>
            </div>
            <div
              class="filterOptionRadio"
            >
              <label
                for="12"
              >
                16+
                <input
                  name="16"
                  type="radio"
                  value="16"
                />
              </label>
            </div>
            <div
              class="filterOptionRadio"
            >
              <label
                for="12"
              >
                18+
                <input
                  name="16"
                  type="radio"
                  value="18"
                />
              </label>
            </div>
          </div>
        </div>
        <div
          class="wrapper"
        >
          <div
            class="productsWrapper hidden"
          >
            <h2
              class="headline"
            >
              Search: 
            </h2>
            <div
              class="productsList"
            >
              <h2
                class="headline"
              >
                Product not found
              </h2>
            </div>
          </div>
          <h2
            class="headline"
          >
            All Products
          </h2>
          <div
            class="productsList"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Products component tests Checks if products page matches recent snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="ldsRoller"
        >
          <div />
          <div />
          <div />
          <div />
          <div />
          <div />
          <div />
          <div />
        </div>
        <div
          class="searchForm"
        >
          <h2
            class="searchFormHeadline"
          >
            Search by name:
          </h2>
          <input
            class="searchInput"
            placeholder="Search..."
            type="text"
          />
        </div>
        <div
          class="body"
        >
          <div
            class="filtration"
          >
            <h1
              class="headline"
            >
              Filter Products
            </h1>
            <div
              class="filtrationItem"
            >
              <h2
                class="headline"
              >
                Price: 
              </h2>
              <button
                class="filterOption"
                type="button"
              >
                ASC
              </button>
              <button
                class="filterOption"
                type="button"
              >
                DESC
              </button>
            </div>
            <div
              class="filtrationItem"
            >
              <h2
                class="headline"
              >
                Rating: 
              </h2>
              <button
                class="filterOption"
                type="button"
              >
                ASC
              </button>
              <button
                class="filterOption"
                type="button"
              >
                DESC
              </button>
            </div>
            <div
              class="filtrationItem"
            >
              <h2
                class="headline"
              >
                Genre: 
              </h2>
              <button
                class="filterOption"
                type="button"
              >
                Action
              </button>
              <button
                class="filterOption"
                type="button"
              >
                Shooter
              </button>
              <button
                class="filterOption"
                type="button"
              >
                All
              </button>
            </div>
            <div
              class="filtrationItem"
            >
              <h2
                class="headline"
              >
                Age: 
              </h2>
              <div
                class="filterOptionRadio"
              >
                <label
                  for="12"
                >
                  12+
                  <input
                    name="12"
                    type="radio"
                    value="12"
                  />
                </label>
              </div>
              <div
                class="filterOptionRadio"
              >
                <label
                  for="12"
                >
                  16+
                  <input
                    name="16"
                    type="radio"
                    value="16"
                  />
                </label>
              </div>
              <div
                class="filterOptionRadio"
              >
                <label
                  for="12"
                >
                  18+
                  <input
                    name="16"
                    type="radio"
                    value="18"
                  />
                </label>
              </div>
            </div>
          </div>
          <div
            class="wrapper"
          >
            <div
              class="productsWrapper hidden"
            >
              <h2
                class="headline"
              >
                Search: 
              </h2>
              <div
                class="productsList"
              >
                <h2
                  class="headline"
                >
                  Product not found
                </h2>
              </div>
            </div>
            <h2
              class="headline"
            >
              All Products
            </h2>
            <div
              class="productsList"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="ldsRoller"
      >
        <div />
        <div />
        <div />
        <div />
        <div />
        <div />
        <div />
        <div />
      </div>
      <div
        class="searchForm"
      >
        <h2
          class="searchFormHeadline"
        >
          Search by name:
        </h2>
        <input
          class="searchInput"
          placeholder="Search..."
          type="text"
        />
      </div>
      <div
        class="body"
      >
        <div
          class="filtration"
        >
          <h1
            class="headline"
          >
            Filter Products
          </h1>
          <div
            class="filtrationItem"
          >
            <h2
              class="headline"
            >
              Price: 
            </h2>
            <button
              class="filterOption"
              type="button"
            >
              ASC
            </button>
            <button
              class="filterOption"
              type="button"
            >
              DESC
            </button>
          </div>
          <div
            class="filtrationItem"
          >
            <h2
              class="headline"
            >
              Rating: 
            </h2>
            <button
              class="filterOption"
              type="button"
            >
              ASC
            </button>
            <button
              class="filterOption"
              type="button"
            >
              DESC
            </button>
          </div>
          <div
            class="filtrationItem"
          >
            <h2
              class="headline"
            >
              Genre: 
            </h2>
            <button
              class="filterOption"
              type="button"
            >
              Action
            </button>
            <button
              class="filterOption"
              type="button"
            >
              Shooter
            </button>
            <button
              class="filterOption"
              type="button"
            >
              All
            </button>
          </div>
          <div
            class="filtrationItem"
          >
            <h2
              class="headline"
            >
              Age: 
            </h2>
            <div
              class="filterOptionRadio"
            >
              <label
                for="12"
              >
                12+
                <input
                  name="12"
                  type="radio"
                  value="12"
                />
              </label>
            </div>
            <div
              class="filterOptionRadio"
            >
              <label
                for="12"
              >
                16+
                <input
                  name="16"
                  type="radio"
                  value="16"
                />
              </label>
            </div>
            <div
              class="filterOptionRadio"
            >
              <label
                for="12"
              >
                18+
                <input
                  name="16"
                  type="radio"
                  value="18"
                />
              </label>
            </div>
          </div>
        </div>
        <div
          class="wrapper"
        >
          <div
            class="productsWrapper hidden"
          >
            <h2
              class="headline"
            >
              Search: 
            </h2>
            <div
              class="productsList"
            >
              <h2
                class="headline"
              >
                Product not found
              </h2>
            </div>
          </div>
          <h2
            class="headline"
          >
            All Products
          </h2>
          <div
            class="productsList"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Products component tests Makes a products snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="ldsRoller"
        >
          <div />
          <div />
          <div />
          <div />
          <div />
          <div />
          <div />
          <div />
        </div>
        <div
          class="searchForm"
        >
          <h2
            class="searchFormHeadline"
          >
            Search by name:
          </h2>
          <input
            class="searchInput"
            placeholder="Search..."
            type="text"
          />
        </div>
        <div
          class="body"
        >
          <div
            class="filtration"
          >
            <h1
              class="headline"
            >
              Filter Products
            </h1>
            <div
              class="filtrationItem"
            >
              <h2
                class="headline"
              >
                Price: 
              </h2>
              <button
                class="filterOption"
                type="button"
              >
                ASC
              </button>
              <button
                class="filterOption"
                type="button"
              >
                DESC
              </button>
            </div>
            <div
              class="filtrationItem"
            >
              <h2
                class="headline"
              >
                Rating: 
              </h2>
              <button
                class="filterOption"
                type="button"
              >
                ASC
              </button>
              <button
                class="filterOption"
                type="button"
              >
                DESC
              </button>
            </div>
            <div
              class="filtrationItem"
            >
              <h2
                class="headline"
              >
                Genre: 
              </h2>
              <button
                class="filterOption"
                type="button"
              >
                Action
              </button>
              <button
                class="filterOption"
                type="button"
              >
                Shooter
              </button>
              <button
                class="filterOption"
                type="button"
              >
                All
              </button>
            </div>
            <div
              class="filtrationItem"
            >
              <h2
                class="headline"
              >
                Age: 
              </h2>
              <div
                class="filterOptionRadio"
              >
                <label
                  for="12"
                >
                  12+
                  <input
                    name="12"
                    type="radio"
                    value="12"
                  />
                </label>
              </div>
              <div
                class="filterOptionRadio"
              >
                <label
                  for="12"
                >
                  16+
                  <input
                    name="16"
                    type="radio"
                    value="16"
                  />
                </label>
              </div>
              <div
                class="filterOptionRadio"
              >
                <label
                  for="12"
                >
                  18+
                  <input
                    name="16"
                    type="radio"
                    value="18"
                  />
                </label>
              </div>
            </div>
          </div>
          <div
            class="wrapper"
          >
            <div
              class="productsWrapper hidden"
            >
              <h2
                class="headline"
              >
                Search: 
              </h2>
              <div
                class="productsList"
              >
                <h2
                  class="headline"
                >
                  Product not found
                </h2>
              </div>
            </div>
            <h2
              class="headline"
            >
              All Products
            </h2>
            <div
              class="productsList"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="ldsRoller"
      >
        <div />
        <div />
        <div />
        <div />
        <div />
        <div />
        <div />
        <div />
      </div>
      <div
        class="searchForm"
      >
        <h2
          class="searchFormHeadline"
        >
          Search by name:
        </h2>
        <input
          class="searchInput"
          placeholder="Search..."
          type="text"
        />
      </div>
      <div
        class="body"
      >
        <div
          class="filtration"
        >
          <h1
            class="headline"
          >
            Filter Products
          </h1>
          <div
            class="filtrationItem"
          >
            <h2
              class="headline"
            >
              Price: 
            </h2>
            <button
              class="filterOption"
              type="button"
            >
              ASC
            </button>
            <button
              class="filterOption"
              type="button"
            >
              DESC
            </button>
          </div>
          <div
            class="filtrationItem"
          >
            <h2
              class="headline"
            >
              Rating: 
            </h2>
            <button
              class="filterOption"
              type="button"
            >
              ASC
            </button>
            <button
              class="filterOption"
              type="button"
            >
              DESC
            </button>
          </div>
          <div
            class="filtrationItem"
          >
            <h2
              class="headline"
            >
              Genre: 
            </h2>
            <button
              class="filterOption"
              type="button"
            >
              Action
            </button>
            <button
              class="filterOption"
              type="button"
            >
              Shooter
            </button>
            <button
              class="filterOption"
              type="button"
            >
              All
            </button>
          </div>
          <div
            class="filtrationItem"
          >
            <h2
              class="headline"
            >
              Age: 
            </h2>
            <div
              class="filterOptionRadio"
            >
              <label
                for="12"
              >
                12+
                <input
                  name="12"
                  type="radio"
                  value="12"
                />
              </label>
            </div>
            <div
              class="filterOptionRadio"
            >
              <label
                for="12"
              >
                16+
                <input
                  name="16"
                  type="radio"
                  value="16"
                />
              </label>
            </div>
            <div
              class="filterOptionRadio"
            >
              <label
                for="12"
              >
                18+
                <input
                  name="16"
                  type="radio"
                  value="18"
                />
              </label>
            </div>
          </div>
        </div>
        <div
          class="wrapper"
        >
          <div
            class="productsWrapper hidden"
          >
            <h2
              class="headline"
            >
              Search: 
            </h2>
            <div
              class="productsList"
            >
              <h2
                class="headline"
              >
                Product not found
              </h2>
            </div>
          </div>
          <h2
            class="headline"
          >
            All Products
          </h2>
          <div
            class="productsList"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
